Objective-C/CBLDatabase.mm

```diff
@@ -722,45 +722,92 @@ - (nullable CBLScope*) scopeWithName: (NSString*)name error: (NSError**)error {
     return nil;
 }
 
-#pragma mark -- Collections
+#pragma mark - Collections
 
 - (nullable CBLCollection*) defaultCollection: (NSError**)error {
-    // TODO: add implementation
-    return  [[CBLCollection alloc] initWithName: kCBLDefaultCollectionName
-                                          scope: [self defaultScope: error]
-                                          error: nil];
+    return [self collectionWithName: kCBLDefaultCollectionName
+                              scope: kCBLDefaultScopeName
+                              error: error];
 }
 
 - (nullable  NSArray*) collections: (nullable NSString*)scope error: (NSError**)error {
-    // TODO: add implementation
-    return [NSArray array];
+    NSString * scopeName = scope ?: kCBLDefaultScopeName;
+    CBLStringBytes sName(scopeName);
+    FLMutableArray list = c4db_collectionNames(_c4db, sName);
+    
+    // TODO: check direct convertion of list to NSArray or each item to NSString is faster?
+    NSUInteger count = FLArray_Count(list);
+    NSMutableArray* collections = [NSMutableArray arrayWithCapacity: count];
+    for (uint i = 0; i < count; i++) {
+        NSString* name = FLValue_GetNSObject(FLArray_Get(list, i), nullptr);
+        
+        CBLCollection* c = [self collectionWithName: name scope: scopeName error: error];
+        [collections addObject: c];
+    }
+    FLArray_Release(list);
+    
+    return [NSArray arrayWithArray: collections];
 }
 
-- (nullable CBLCollection*) createCollectionWithName: (NSString*)name
-                                      scope: (nullable NSString*)scope
-                                      error: (NSError**)error {
+- (nullable CBLCollection*) createCollectionWithName: (NSString*)collectionName
+                                               scope: (nullable NSString*)scopeName
+                                               error: (NSError**)error {
+    scopeName = scopeName.length > 0 ? scopeName : kCBLDefaultScopeName;
```

Comment:
Just check nil scopeName for the default scope. If using nil, LiteCore will return an error when creating the collection.